components/automate-ui/src/app/page-components/projects-filter-dropdown/projects-filter-dropdown.spec.ts

```diff
@@ -174,6 +174,277 @@ describe('ProjectsFilterDropdownComponent', () => {
+        component.handleOptionChange({ detail: true }, 'Project 1');
+        component.handleOptionChange({ detail: false }, 'Project 2');
+        component.handleOptionChange({ detail: false }, 'Project 3');
+        expect(component.optionsEdited).toBe(true);
+
+        component.handleOptionChange({ detail: false }, 'Project 1');
+        component.handleOptionChange({ detail: true }, 'Project 2');
+        component.handleOptionChange({ detail: true }, 'Project 3');
+        expect(component.optionsEdited).toBe(false);
+      });
+
+      it('check then uncheck all other projects', () => {
+        component.handleOptionChange({ detail: true }, 'Project 1');
+        component.handleOptionChange({ detail: true }, 'Project 4');
+        component.handleOptionChange({ detail: true }, 'Project 5');
+        expect(component.optionsEdited).toBe(true);
+
+        component.handleOptionChange({ detail: false }, 'Project 1');
+        component.handleOptionChange({ detail: false }, 'Project 4');
+        component.handleOptionChange({ detail: false }, 'Project 5');
+        expect(component.optionsEdited).toBe(false);
+      });
+    });
+
+    describe('with all projects checked...', () => {
+      beforeEach(() => {
+        component.editableOptions = genOptions([true, true, true, true, true]);
+        component.filteredOptions = component.editableOptions;
+        component.initialFilters = component.filteredOptions.map(option => option.checked);
+      });
+
+      it('we uncheck then check one project', () => {
+        component.handleOptionChange({ detail: false }, 'Project 1');
+        expect(component.optionsEdited).toBe(true);
+
+        component.handleOptionChange({ detail: true }, 'Project 1');
+        expect(component.optionsEdited).toBe(false);
+      });
+
+      it('we uncheck then check multiple projects', () => {
+        component.handleOptionChange({ detail: false }, 'Project 1');
+        component.handleOptionChange({ detail: false }, 'Project 4');
+        component.handleOptionChange({ detail: false }, 'Project 5');
+        expect(component.optionsEdited).toBe(true);
+
+        component.handleOptionChange({ detail: true }, 'Project 1');
+        component.handleOptionChange({ detail: true }, 'Project 4');
+        component.handleOptionChange({ detail: true }, 'Project 5');
+        expect(component.optionsEdited).toBe(false);
+      });
+
+      it('we uncheck then recheck all projects', () => {
+        component.editableOptions.forEach(
+          (_o, index) => component.handleOptionChange({ detail: false }, `Project ${index + 1}`));
+        expect(component.optionsEdited).toBe(true);
+
+        component.editableOptions.forEach(
+          (_o, index) => component.handleOptionChange({ detail: true }, `Project ${index + 1}`));
+        expect(component.optionsEdited).toBe(false);
+      });
+    });
+
+    describe('when options have filters applied', () => {
```

Comment:
:+1: I would agree with the behavior described in this test block. Changed options should enable the Apply button regardless if they are filtered afterwards.

Good feedback @msorens ensuring we specify behavior here.

Reply:
Thanks @scottopherson - I appreciate the feedback!